### PR TITLE
Data Structure session 8 Assignment

### PR DESCRIPTION
**Pull request by meta-udit-saxena**

### CODE REVIEW

File: DataStructure 8 assignment/DataStructure8-Q1/src/com/metacube/testcase/BinarySearchTreeTestCases.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package com.metacube.testcase;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import com.metacube.binarytree.BinarySearchTree;
+import com.metacube.entity.Student;
+
+public class BinarySearchTreeTestCases {
```
Comment (chirag-metacube) [inline DataStructure 8 assignment/DataStructure8-Q1/src/com/metacube/testcase/BinarySearchTreeTestCases.java:9]: More test cases can be written here, including null values and other methods

File: DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/controller/SortingSystemController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	 * if input number of element is greater than 10 than it perform quick sort
+	 * else bubble sort
+	 *
+	 * @param inputArray
+	 *            the input array
+	 */
+	public SortResponse doComparisonSort(int[] inputArray) {
+		SortResponse response = ComparisonSortFacade.getInstance().sort(
+				inputArray);
+		new Main().displayResponse(response);
```
Comment (chirag-metacube) [inline DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/controller/SortingSystemController.java:24]: Controller should not call the Main class functions

File: DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/testcases/DoLinearSortTestCases.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	SortingSystemController controller;
+
+	@Before
+	public void setUpObject() {
+		controller = new SortingSystemController();
+	}
+
+	@Parameters
+	public static Collection<Object[]> insert() {
+		return Arrays.asList(new Object[][]{
```
Comment (chirag-metacube) [inline DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/testcases/DoLinearSortTestCases.java:28]: Good

File: DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/utility/Utility.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+public class Utility {
+
+	/**
+	 * Gets the lowest value.
+	 *
+	 * @param array
+	 *            the array
+	 * @return the lowest value
+	 */
+	public static Integer getLowestValue(int[] array) {
```
Comment (chirag-metacube) [inline DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/utility/Utility.java:12]: it can just return 'int' instead of 'Integer'

File: DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/utility/Utility.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	 *
+	 * @param array
+	 *            the array
+	 * @param index1
+	 *            the index 1
+	 * @param index2
+	 *            the index 2
+	 */
+	public static void swap(int[] array, int index1, int index2) {
+		int temp = array[index1];
```
Comment (chirag-metacube) [inline DataStructure 8 assignment/DataStructure8-Q2/src/com/metacube/sortingsystem/utility/Utility.java:59]: Nice

**Review by chirag-metacube — APPROVED** [review]
